spec/recipe.fmf

```diff
@@ -0,0 +1,87 @@
+story: |
+    As a user, I want tmt to generate a recipe file that can be reused to reproduce
+    a run with the exact same configuration, without requiring additional input.
+
+description: |
+    A recipe is a YAML file that captures a complete, static snapshot of a tmt
+    run after all dynamic evaluation has been resolved. It includes preprocessed
+    information about plans, tests, results, and run configuration. All environment
+    variables are evaluated, and git references are replaced with fixed commit hashes.
+
+    .. code-block:: yaml
+
+        # Run  configuration
+        run:
+            root: "/path/to/fmf/root"
+            remove: false
```

Comment:
In order to be able to reproduce the run we need the git repo details as well. It should be basically fmf id, just without the name, so `url` of the repo, `ref` in the form of a hash and `path` to  the metadata tree. The `name` part should be covered by plan names under the `plans` key below.

Also, I guess, we should store here the original `environment` and `context` with which the run was created? The complete `environment` should be stored for each test in `tests.yaml`, but the environment variables are also delivered to other places, such as scripts executed during the `prepare` step.